serial/serial_resources/component_instance.py

```diff
@@ -0,0 +1,112 @@
+                                                                params=child_component_instance_params)[0]
+        link_dataset = self.client.make_api_request("/datasets",
+                                                    "GET",
+                                                    params=link_params)[0]
+        process_entry_id = None
+        if process_entry:
+            process_entry_id = process_entry.id
+        else:
+            process_entry_id = self.process_entry.id
+        data = {
+                "parent_component_instance_id": self.data["id"],
+                "child_component_instance_id": child_component_instance["id"],
+                "dataset_id": link_dataset["dataset"]["id"],
+                "process_entry_id": process_entry_id,
+                "break_prior_links": break_prior_links,
+                }
+        self.links.append(ComponentInstanceLink(self.client.make_api_request("/components/instances/links", "PUT", data=data)))
+        return self.links[-1]
+
+class ComponentInstances:
+    def __init__(self):
+        pass
+
+    @staticmethod
+    def get(identifier):
+        """
+        Gets a component instance, if it exists
+        
+        Args:
+        - identifier: Component's user facing identifier
+
+        Returns:
+        - A component instance object, as defined at 
+        https://docs.serial.io/api-reference/component-instances/get-component-instance
+        """
+        client = APIClient(serial.api_key, serial.base_url) 
+        if serial.debug:
+            print(f"Getting component instance: {identifier}")
+        return ComponentInstance(client.make_api_request(f"/components/instances/{identifier}", "GET"))
+    
+    @staticmethod
+    def create(identifier, component_name):
+        """
+        Creates a component instance
+        
+        Args:
+        - identifier: Component's user facing identifier
+        - component_name: User facing name for the component
+            
+        Returns:
+        - A component instance, as defined at 
+        https://docs.serial.io/api-reference/component-instances/get-component-instance
+        """
+        if serial.debug:
+            print(f"Creating component instance: {identifier} with component name: {component_name}")
+        client = APIClient(serial.api_key, serial.base_url) 
+        # Get the component id for the given component name
+        component_name_params = {
+                "name" : component_name,
+                }
+        component_id = client.make_api_request("/components",
+                                                     "GET",
+                                                     params=component_name_params)[0]["id"]
+        data = {
+                "component_id": component_id,
+                "identifier": identifier,
+                }
+        return ComponentInstance(client.make_api_request(f"/components/instances", "PUT", data=data)) 
+
+    @staticmethod
+    def list(query_params):
```

Comment:
I'm curious what this looks like? how do query params get passed?

Reply:
Query params are simply a set of parameters (i.e. a dictionary) that gets turned into a `url?param1=a&param2=b` format